src/subscription_manager/api/__init__.py

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+"""
+Class and methods made available in __all__ are provided for use by
+external applications.
+
+All reasonable efforts will be made to maintain compatibility.
+"""
+from functools import wraps
+
+injected = False
+
+
+def request_injection(func):
```

Comment:
Doc string describing when it's appropriate to use this and what it does?

(injection singleton for plugins?)